pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -197,7 +198,7 @@ func convertHTTPRoute(r k8s.HTTPRouteRule, ctx configContext,
 	vs := &istio.HTTPRoute{}
 	// Auto-name the route. If upstream defines an explicit name, will use it instead
 	// The position within the route is unique
-	vs.Name = fmt.Sprintf("%s.%s.%d", obj.Namespace, obj.Name, pos)
+	vs.Name = obj.Namespace + "." + obj.Name + "." + strconv.Itoa(pos) // formt: %s.%s.%d
```

Comment:
nit; typo in comment:
```suggestion
	vs.Name = obj.Namespace + "." + obj.Name + "." + strconv.Itoa(pos) // format: %s.%s.%d
```